### PR TITLE
7857: Apply moduleEditor feedback

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
On Known Modules page, 'Edit Module' link should only be shown if moduleEditor is not included in the build, or if the relevant module is not editable.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/345
* https://github.com/LabKey/moduleEditor/pull/12

#### Changes
* Add helper function to DefaultModule
* Add checks for moduleEditor's existence and module validity before displaying 'Edit Module' link

### CODE REVIEW

File: core/src/org/labkey/core/admin/AdminController.java [hunk 1]
```diff
@@ -8111,6 +8111,8 @@ protected void renderView(Object model, PrintWriter out)
                                         if (!rel.startsWith(".."))
                                             shortPathToModule = rel.toString();
                                     }
+                                    boolean modIsValid = (module instanceof DefaultModule) && ((DefaultModule) module).isValidSourceDirectory();
```
Comment (RosalineP) [inline core/src/org/labkey/core/admin/AdminController.java]: @labkey-matthewb 
This looked like the correct check to invoke based off of your comments on the ModuleEditor spec regarding when to allow editing of module resources. Happy to make any fixes if I've misapprehended anything, or if there are additional gates to include when determining if a module is editable or not!

Comment (labkey-matthewb) [reply core/src/org/labkey/core/admin/AdminController.java]: This logic shouldn't be scattered around the product, it should be centralized in ModuleEditorService.  Presumably null != ModuleService.getModuleEditorURL() should be accurate.

**Review by labkey-matthewb — CHANGES_REQUESTED** [review]

File: core/src/org/labkey/core/admin/AdminController.java (hunk 1 above)
Comment (RosalineP) [reply core/src/org/labkey/core/admin/AdminController.java]: Hopefully addressed as per offline gChat convo.

Comment (labkey-matthewb) [reply core/src/org/labkey/core/admin/AdminController.java]: Doesn't address the original comment however.  Presumably getModuleEditorURL() is correctly determining if this module can be edited and what the URL should be.  Why second guess that code???  Just check != null.

**Review by labkey-matthewb — CHANGES_REQUESTED** [review]

File: core/src/org/labkey/core/admin/AdminController.java (hunk 1 above)
Comment (RosalineP) [reply core/src/org/labkey/core/admin/AdminController.java]: Ah, the intention of the `isValidSourceDirectory()` check goes back to a comment you made on the ModuleEditor spec (under the 'Open questions' section) regarding when to allow editing of module resources—you mention that we should allow editing if we have a valid source directory, and to take a look at logic implemented in `computeResourceDirectory()` for that. I thought that this was what you were referring to when you mention in the Issue opened that if a module isn't editable, the link to the ModuleEditor shouldn't appear. (My bad—should have made this more clear than my vague comment above about the ModuleEditor spec!)

It sounds like I was probably misinterpreting that, though, and the nullcheck with getModuleEditorURL() is sufficient here for determining whether to display the link to the ModuleEditor on the known modules page! Changes pushed.

**Review by labkey-matthewb — APPROVED** [review]

**Pull request merged**